applications/RANSApplication/custom_utilities/rans_variable_utilities.cpp

```diff
@@ -230,6 +234,239 @@ void CopyNodalSolutionStepVariablesList(
+                r_node.FastGetSolutionStepValue(rVariable) +=
+                    r_normal * (1.0 / static_cast<double>(number_of_nodes));
+                r_node.UnSetLock();
+            }
+        }
+    }
+
+    rModelPart.GetCommunicator().AssembleCurrentData(rVariable);
+}
+
+void AddAnalysisStep(ModelPart& rModelPart, const std::string& rStepName)
+{
+    auto& r_process_info = rModelPart.GetProcessInfo();
+    if (!r_process_info.Has(ANALYSIS_STEPS))
+    {
+        r_process_info.SetValue(ANALYSIS_STEPS, std::vector<std::string>());
+    }
+    r_process_info[ANALYSIS_STEPS].push_back(rStepName);
+}
+
+bool IsAnalysisStepCompleted(const ModelPart& rModelPart, const std::string& rStepName)
+{
+    const auto& r_process_info = rModelPart.GetProcessInfo();
+    if (r_process_info.Has(ANALYSIS_STEPS))
+    {
+        const std::vector<std::string>& r_steps = r_process_info[ANALYSIS_STEPS];
+        return (std::find(r_steps.begin(), r_steps.end(), rStepName) != r_steps.end());
+    }
+    else
+    {
+        return false;
+    }
+}
+
+void AssignBoundaryFlagsToGeometries(ModelPart& rModelPart)
+{
+    const int number_of_conditions = rModelPart.NumberOfConditions();
+#pragma omp parallel for
+    for (int i_cond = 0; i_cond < number_of_conditions; ++i_cond)
+    {
+        Condition& r_condition = *(rModelPart.ConditionsBegin() + i_cond);
+        r_condition.SetValue(RANS_IS_INLET, r_condition.Is(INLET));
+        r_condition.SetValue(RANS_IS_OUTLET, r_condition.Is(OUTLET));
+        r_condition.SetValue(RANS_IS_STRUCTURE, r_condition.Is(STRUCTURE));
+    }
+}
+
+void FixFlaggedDofs(ModelPart& rModelPart,
+                    const Variable<double>& rFixingVariable,
+                    const Flags& rFlag,
+                    const bool CheckValue)
+{
+    KRATOS_TRY
+
+    const int number_of_nodes = rModelPart.NumberOfNodes();
+#pragma omp parallel for
+    for (int i_node = 0; i_node < number_of_nodes; ++i_node)
+    {
+        auto& r_node = *(rModelPart.NodesBegin() + i_node);
+        if (r_node.Is(rFlag) == CheckValue)
+        {
+            r_node.Fix(rFixingVariable);
+        }
+    }
+
+    KRATOS_CATCH("");
+}
+
+template <typename TDataType>
+void CopyNodalFlaggedVariableFromNonHistoricalToHistorical(ModelPart& rModelPart,
+                                                           const Variable<TDataType>& rVariable,
+                                                           const Flags& rFlag,
+                                                           const bool CheckValue)
+{
+    KRATOS_TRY
+
+    const int number_of_nodes = rModelPart.NumberOfNodes();
+#pragma omp parallel for
+    for (int i_node = 0; i_node < number_of_nodes; ++i_node)
+    {
+        auto& r_node = *(rModelPart.NodesBegin() + i_node);
+        if (r_node.Is(rFlag) == CheckValue)
+        {
+            r_node.FastGetSolutionStepValue(rVariable) = r_node.GetValue(rVariable);
+        }
+    }
+
+    KRATOS_CATCH("");
+}
```

Comment:
Same comment. We can enhance the methods in the `VariableUtils` that do this operations to be flag compatible.

Reply:
done in #7302